src/casts/specify-representation.rkt

```diff
@@ -1236,90 +1213,112 @@ but a static single assignment is implicitly maintained.
         [(Type-Tuple-num (app recur e)) (type-tuple-count-access e)]
         [(Type-Tuple-item (app recur e) (app recur i))
          (type-tuple-elements-access e i)]
-        [(Coerce-Tuple uid (app recur v) (app recur c))
-         (: invoke-coerce-tuple ((Code-Label Uid) (Var Uid) D0-Expr -> D0-Expr))
-         (define (invoke-coerce-tuple coerce-tuple v c)
-           (begin$
-             (assign$ tagged-count
-               (sr-tagged-array-ref
-                c COERCION-MEDIATING-TAG COERCION-TUPLE-COUNT-INDEX))
-             (App-Code
-              coerce-tuple
-              (list v c ZERO-IMDT
-                    (op$ %>> tagged-count COERCION-SECOND-TAG-SHIFT)))))
-         (let ([coerce-tuple (get-coerce-tuple! uid)])
-           (if (Var? v)
-               (invoke-coerce-tuple coerce-tuple v c)
-               (begin$
-                 (assign$ tuple-val1 v)
-                 (invoke-coerce-tuple coerce-tuple tuple-val1 c))))]
-        [(Coerce-Tuple-In-Place uid (app recur v) (app recur c) (app recur mono-type))
-         (: invoke-coerce-tuple ((Code-Label Uid) (Var Uid) D0-Expr D0-Expr -> D0-Expr))
-         (define (invoke-coerce-tuple coerce-tuple v c a)
-           (App-Code
-            coerce-tuple
-            (list v c a)))
-         (let ([coerce-tuple (get-coerce-tuple-in-place! uid)])
-           (if (Var? v)
-               (invoke-coerce-tuple coerce-tuple v c mono-type)
-               (begin$
-                 (assign$ tuple-val1 v)
-                 (invoke-coerce-tuple coerce-tuple tuple-val1 c mono-type))))]
+        [(Coerce-Tuple coerce (app recur v) (app recur c))
+         (define coerce-tuple (get-coerce-tuple! coerce))
+         (: invoke-coerce-tuple ((Var Uid) (Var Uid) -> D0-Expr))
+         (define (invoke-coerce-tuple v c)
+           (app-code$ coerce-tuple v c))
+         (cond
+           [(and (Var? v) (Var? c)) (invoke-coerce-tuple v c)]
+           [(Var? v) (begin$ (assign$ a-c c) (invoke-coerce-tuple v a-c))]
+           [(Var? c) (begin$ (assign$ a-v v) (invoke-coerce-tuple a-v c))]
+           [else (begin$
+                   (assign$ a-c c)
+                   (assign$ a-v v)
+                   (invoke-coerce-tuple a-v a-c))])]
+        [(Coerce-Tuple-In-Place coerce (app recur v) (app recur c) (app recur mono-type))
+         (define coerce-tuple (get-coerce-tuple-in-place! coerce))
+         (: invoke-coerce-tuple ((Var Uid) (Var Uid) -> D0-Expr))
+         (define (invoke-coerce-tuple v c)
+           (app-code$ coerce-tuple v c mono-type))
+         (cond
+           [(and (Var? v) (Var? c)) (invoke-coerce-tuple v c)]
+           [(Var? v) (begin$ (assign$ a-c c) (invoke-coerce-tuple v a-c))]
+           [(Var? c) (begin$ (assign$ a-v v) (invoke-coerce-tuple a-v c))]
+           [else (begin$
+                   (assign$ a-c c)
+                   (assign$ a-v v)
+                   (invoke-coerce-tuple a-v a-c))])]
```

Comment:
This chunk of code, that is doing manual ANF, doesn't seem to be necessary.

Reply:
Done.